src/components/SQForm/SQFormMultiSelect.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Select from '@material-ui/core/Select';
+import Input from '@material-ui/core/Input';
+import MenuItem from '@material-ui/core/MenuItem';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Grid from '@material-ui/core/Grid';
+import Checkbox from '@material-ui/core/Checkbox';
+import ListItemText from '@material-ui/core/ListItemText';
+import {useSQFormContext} from '../../../src';
+
+import {useForm} from './useForm';
+
+function SQFormMultiSelect({
+  children,
+  isDisabled = false,
+  isRequired = false,
+  label,
+  name,
+  size = 'auto',
+  useSelectAll = true
+}) {
+  const {setFieldValue} = useSQFormContext();
+  const {
+    formikField: {field},
+    fieldState: {isFieldError},
+    fieldHelpers: {handleBlur, HelperTextComponent}
+  } = useForm({
+    name,
+    isRequired
+  });
+  const labelID = label.toLowerCase();
+
+  const handleMultiSelectChange = event => {
+    const selectAllWasChecked = event.target.value.includes('ALL');
+    const selectNoneWasChecked = event.target.value.includes('NONE');
+
+    const values = selectAllWasChecked
```

Comment:
I'm all for a better approach as to how we handle what gets sent to the `setFieldValue` function -- but I do like the fact that I can just use the children prop passed in to generate the list of things to select when I hit "Select All"

Reply:
can we refactor this a bit to avoid the nested ternary? I like the idea, generally, but I think if we wrote a little helper function or two we could make it a bit more readable.

Reply:
Yeah if the nested ternary is a no-go, I guess I could just write that out in if statements instead? It's a simple enough conditional that I don't know if a helper function would do much other than abstract the logic away from the function.

Reply:
yeah that works for me

Reply:
I think a functional approach would alleviate the need for the nested logic

```javascript
const value = event.target.value;
const isSelectAllChecked = getIsSelectAllChecked(value); // return a boolean
const values = getValues(children, isSelectAllChecked, value) // return children.map(), [], or value

setFieldValue(name, values);
```